metricbeat/mb/module/event.go

```diff
@@ -68,37 +59,32 @@ func (b EventBuilder) Build() (common.MapStr, error) {
 	if n, ok := event["_namespace"]; ok {
 		delete(event, "_namespace")
 		namespace = n.(string)
-		// TODO: check if namespace does not already exist
 		metricsetData["namespace"] = namespace
 	}
 
-	event = common.MapStr{
-		"@timestamp":            timestamp,
-		"type":                  typeName,
-		common.EventMetadataKey: b.metadata,
-		b.ModuleName: common.MapStr{
-			namespace: event,
-		},
-		"metricset": metricsetData,
+	// Checks if additional meta information is provided by the MetricSet under the key ModuleData
+	// This is based on the convention that each MetricSet can provide module data under the key ModuleData
+	moduleData, moudleDataExists := event[mb.ModuleDataKey]
+	if moudleDataExists {
+		delete(event, mb.ModuleDataKey)
 	}
 
+	moduleEvent := common.MapStr{}
+	moduleEvent.Put(namespace, event)
+
 	// In case meta data exists, it is added on the module level
+	// This is mostly used for shared fields across multiple metricsets in one module
 	if moudleDataExists {
 		if _, ok := moduleData.(common.MapStr); ok {
```

Comment:
Instead of ignoring the result of the type assertion with an assignment to `_`, we could use the value in the following expression instead of doing a second type assertion.